bot.js

```diff
@@ -218,10 +218,22 @@ function balance(pool, message){
 		groups[0] = [leaders[0], leaders[1], ...pool]
 	} else {
 		message.channel.send("Zakum has detected two groups! Balancing groups...")
-		let difference = 1000;
+		let difference = 1000; // this being used?
 		let g0 = [leaders[0]]
 		let g1 = [leaders[1]]
-		let poolToJoin = pool.filter(person => person !== leaders[0] && person !== leaders[1]).sort((a,b) => b.rank-a.rank)
+		let bishops = 	 pool.filter(person => person.role == 'bishop').sort((a,b) => b.rank-a.rank) 		// gimmie my bishes sorted by rank
+		let poolToJoin = pool.filter(person => !person.leader && person.role.toLowerCase() != 'bishop') // gimmie my non-leaders/non-bishops
+
+		/* 
+			 ALLOCATE BISHOPS
+			 assumptions:
+			 - groups may have existing members in them
+		*/
+		bishops.forEach(function(b){
+			if(!g0.map(g => g.role.toLowerCase()).includes('bishop')){ g0.push(b); }	// give g0 a bish if they don't have one
+			else if(!g1.map(g => g.role).includes('bishop')){ g1.push(b) } // give g1 a bish if they don't have one
```

Comment:
Missing g.role.toLowerCase() here?

Reply:
Will g0 always get the best Bishop every time with this logic?

Reply:
Would you prefer to get the weaker bishop?

Reply:
I won't always be a raid leader

Reply:
I think the leader ranks need to be compared here